### PR TITLE
Accumulate chunks until min size (1Mb) is reached

### PR DESCRIPTION
**Pull request by kramarmm**
Closes: https://github.com/metarhia/metacom/issues/326

- [x] code is properly formatted (`npm run fmt`)

### CODE REVIEW

File: dist/streams.js [hunk 1]
```diff
@@ -8,10 +8,7 @@ const finisherByte = 59; // ;
 class MetacomChunk {
   static encode(streamId, payload) {
     const metadata = encoder.encode(`mc:${streamId};`);
-    const byteView = new Uint8Array(metadata.length + payload.length);
-    byteView.set(metadata);
-    byteView.set(payload, metadata.length);
-    return byteView;
+    return this.merge(metadata, payload);
```
Comment (tshemsedinov) [inline dist/streams.js:11]:
```suggestion
    return MetacomChunk.merge(metadata, payload);
```

File: dist/streams.js [hunk 2]
```diff
@@ -32,12 +29,20 @@ class MetacomChunk {
     }
     throw new Error('Invalid chunk metadata: ' + metadata);
   }
+
+  static merge(ch1, ch2) {
+    const result = new Uint8Array(ch1.length + ch2.length);
+    result.set(ch1);
+    result.set(ch2, ch1.length);
+    return result;
+  }
```
Comment (tshemsedinov) [inline dist/streams.js:38]: It's better to move this method out of class as a pure function

File: dist/streams.js [hunk 3]
```diff
@@ -155,6 +160,7 @@ class MetacomWritable extends EventEmitter {
     this.streamId = initData.streamId;
     this.name = initData.name;
     this.size = initData.size;
+    this.chunk = null;
```
Comment (tshemsedinov) [inline dist/streams.js:163]: Maybe it's better to hold chunks in array before concatenation and concatenate them once.

Comment (kramarmm) [reply dist/streams.js]: Done

File: dist/streams.js (hunk 2 above)
Comment (kramarmm) [reply dist/streams.js]: Done

**Comment by kramarmm** [comment]
@tshemsedinov I've applied proposed suggestions.
Could you take a look again?

**Comment by tshemsedinov** [comment]
To sync this fork woth metacom/master please drop first 4 commits and rebase on metacom/master

**Comment by rohiievych** [comment]
Closing in favor of extended version: https://github.com/metarhia/metacom/issues/336
Proceed with the new PR for extended issue.

**Pull request closed without merging**